transformer/task.py

```diff
@@ -203,53 +223,114 @@ def replace_url(self, url: str):
+
+    return rpd
+
+
+def _extract_text(
+    mime: str, post_data: dict, text_k: str, rpd: RequestsPostData
+) -> None:
+    text = post_data.get(text_k)
+    if mime == JSON_MIME_TYPE:
+        if text is None:
+            raise ValueError(f"missing {text_k!r} field for {JSON_MIME_TYPE} content")
+        try:
+            rpd.json = py.Literal(json.loads(text))
+        except JSONDecodeError as err:
+            raise ValueError(f"unreadable JSON from field {text_k!r}") from err
+    elif text is not None:  # Probably application/x-www-form-urlencoded.
+        try:
+            rpd.data = py.Literal(text.encode())
+        except UnicodeEncodeError as err:
+            raise ValueError(f"cannot encode the {text_k!r} field in UTF-8") from err
+
+
+def _params_from_post_data(
+    key: str, post_data: dict
+) -> Optional[List[Tuple[bytes, bytes]]]:
+    """
+    Extracts the *key* list from *post_data* and calls
+    _params_from_name_value_dicts with that list.
 
-        output_string += line.rjust(len(line) + indentation, " ") + "\n"
+    :raise TypeError: if the object at *key* is built using unexpected data types.
+    """
+    params = post_data.get(key)
+    if params is None:
+        return
+    if not isinstance(params, list):
+        raise TypeError(f"the {key!r} field should be a list")
+    return _params_from_name_value_dicts(params)
 
-    return output_string
 
+def _params_from_name_value_dicts(
+    dicts: Iterable[Mapping[str, str]]
+) -> List[Tuple[bytes, bytes]]:
+    """
+    Converts a HAR "params" element [0] into a list of tuples that can be used
+    as value for requests' "params" keyword-argument.
 
-def _parse_post_data(post_data: dict) -> dict:
-    data = post_data.get("text")
-    mime: str = post_data.get("mimeType")
-    if mime == "application/json":
-        key = "json"
-        # Workaround for bug in chrome-har:
-        # https://github.com/sitespeedio/chrome-har/issues/23
-        # TODO: Remove once bug fixed.
-        if data is None:
-            params = post_data.get("params")
-            if params is None:
-                data = ""
-            else:
-                data = {}
-                for param in params:
-                    data[param.get("name")] = param.get("value")
-        else:
-            data = json.loads(data)
-    else:
-        key = "data"
-        if data:
-            data = data.encode()
-    return {"key": key, "data": data}
+    [0]: http://www.softwareishard.com/blog/har-12-spec/#params
+    [1]: http://docs.python-requests.org/en/master/user/quickstart/
+        #more-complicated-post-requests
+
+    :raise KeyError: if one of the elements doesn't contain a "name" or "value" field.
+    :raise UnicodeEncodeError: if an element's "name" or "value" string cannot
+        be encoded in UTF-8.
+    """
+    return [(d["name"].encode(), d["value"].encode()) for d in dicts]
+
+
+JSON_MIME_TYPE = "application/json"
```

Comment:
Please move the constant variable's definition to the top of the module (below imports, where all other constants are defined)